editors/js/threeDVE.js

```diff
@@ -0,0 +1,1329 @@
+ */
+var EDITORS = EDITORS || {};
+
+/**
+ * Constructor for 3D VE Object
+ *
+ * @type Object
+ *
+ * @param config is a JSON object with the following possible properties
+ *
+ * @returns {threeDVE}
+ */
+
+EDITORS.threeDVE = function (config) {
+  var tdVE = this;
+  //read configuration and set defaults
+  this.config = config;
+  this.type = "3DVE";
+  this.editDiv = config['editDiv'] ? config['editDiv'] : null;
+  this.dataMgr = config['dataMgr'] ? config['dataMgr'] : null;
+  this.edition = config['edition'] ? config['edition'] : null;
+  this.ednID = (config['ednID'] && this.dataMgr &&
+      this.dataMgr.entities.edn[config['ednID']]) ? config['ednID'] : null;
+  this.layoutMgr = config['layoutMgr'] ? config['layoutMgr'] : null;
+  this.id = config['id'] ? config['id'] : (this.editDiv.id ? this.editDiv.id : null);
+  this.api = null;
+  this.annotationData = {};
+  this.viewToolbar = null;
+  this.editToolbar = null;
+  this.toolAnnotateDiv = null;
+  this.toolPropertyDiv = null;
+  this.toolAnnotateBtn = null;
+  this.toolPropertyBtn = null;
+  this.annotateMode = false;
+  this.contentWrapperElement = null;
+  this.propPaneExpanded = false;
+  this.init();
+  return this;
+};
+
+/**
+ * Prototype for 3D VE Object
+ */
+EDITORS.threeDVE.prototype = {
+
+  /**
+   * Initiate the VE object.
+   */
+  init: function () {
+    var tdVE = this;
+    this.contentWrapperElement = $('<div class="tdv-content"></div>');
+    var frameWidth = $(this.editDiv).width();
+    var frameHeight = $(this.editDiv).height();
+    var frameElement = $('<div class="iframe-wrapper"><div class="iframe-status">Adjust the camera and click on the model</div><div class="iframe-overlay"></div><iframe src="" id="tdvFrame" allow="autoplay; fullscreen; vr" ' +
+        'allowvr allowfullscreen mozallowfullscreen="true" webkitallowfullscreen="true">' +
+        '</iframe></div>');
+    var propertyPane = $('<div class="tdv-prop-pane"></div>');
+    propertyPane.css({
+      "padding": '10px',
+      'background-color': '#d9d9d9',
+      'border-left': '2px solid #828181'
+    });
+    this.contentWrapperElement.append(frameElement);
+    this.contentWrapperElement.append(propertyPane);
+    $(this.editDiv).append(this.contentWrapperElement);
+    this.contentWrapperElement.jqxSplitter({
+      width: '100%',
+      height: '100%',
+      orientation: 'vertical',
+      splitBarSize: 1,
+      showSplitBar:false,
+      panels: [
+          { size: '60%', min: '250', collapsible: false},
+          { size: '40%', min: '150', collapsed: true, collapsible: true}
+      ]
+    });
+    this.loadDefaultPropertyView();
+
+    // Initiate the viewer.
+    $(this.editDiv).find('.iframe-wrapper').width(frameWidth);
+    $(this.editDiv).find('.iframe-wrapper').height(frameHeight);
+    $(this.editDiv).find('.iframe-wrapper').css({
+      "display": "inline-block",
+      "position": "relative"
+    });
+    $(this.editDiv).find('.iframe-status').css({
```

Comment:
These styles are required base styles for the VE rather than the theming styles. As there are not many styles used anyway, let's keep it this way for the moment.